### DISCUSSION TITLE
Collection variables support

### DISCUSSION THREAD

**Discussion by davidtimmsinshur** [apideck-libraries/postman-to-k6#127] [Q&A] [answered]
I'd like to clarify whether the tool supports collection variables? It doesn't seem to at the moment.
The features list mentions: 

> Global variables exposed by Postman: globals environment data iteration.

So it's not called out in the supported features but it's also not mentioned in unsupported features section. I'm sure we've used the tool in the past to include collection variables but maybe we had some other workaround that I wasn't aware of (can't find the old example as it's since been refactored).

We're using collection variables because it makes a neat exportable unit that can be imported without loading separate environment variables.

**Comment by thim81** [comment]
hi @davidtimmsinshur 

postman-to-k6 support collection variables.

In the readme, there is a example on how to use them: 
https://github.com/apideck-libraries/postman-to-k6?tab=readme-ov-file#global-variables

Does this help you?

**Comment by davidtimmsinshur** [comment, marked as answer]
Hi, @thim81 

Thanks for getting back to me, appreciate it. I was aware of the capability to export global variables into a file and import them into the script. Collection variables are slightly different because they are not exported like that but as part of the collection. Looking into it further for some reason the collection I was trying to convert didn't include the collection variables in the postman export, presumably my error. I've re-exported the collection and postman-to-k6 converts this and the variables are used when I run the output in k6. So it was probably something I did wrong in the original export.

Cheers

**Reply by thim81** [reply]
Thanks for taking the time to share your findings and resolution.

I haven't used the global variables in a while, so I'll verify the behavior when I have some time and update the readme if needed.

**Reply by davidtimmsinshur** [reply]
Further reading if interested 

https://learning.postman.com/collection-format/advanced-concepts/variables/#collection-variables
and relevant https://learning.postman.com/docs/sending-requests/variables/variables/#variable-scopes

In the interface they are set within the collection config eg
![image](https://github.com/apideck-libraries/postman-to-k6/assets/64795684/b2bb1a69-2a21-45b1-9044-a078f2233e25)


All the best
